File: Backend/Iteramico/DomainRepository/Models/Memory.cs
namespace DomainRepository.Models
{
#nullable disable
    public class Memory
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string ImageBase64 { get; set; }
        public DateTime PostedOn { get; set; }

        public virtual User Creator {  get; set; }
        public virtual Journey Journey { get; set; }
    }
}

File: frontend/lib/models/memory.dart
import 'package:frontend/models/user.dart';

class Memory {
  final int id;
  final String name;
  final String imageBase64;
  final DateTime postedOn;
  final User creator;

  Memory(
      {required this.id,
      required this.name,
      required this.imageBase64,
      required this.postedOn,
      required this.creator});

  factory Memory.fromJson(Map<String, dynamic> json) {
    return Memory(
        id: json['id'],
        name: json['name'],
        imageBase64: json['imageBase64'],
        postedOn: DateTime.parse(json['postedOn']),
        creator: User.fromJson(json['creator']));
  }
}
